pkg/client/checkpointer_test.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2022 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package client
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var path = "checkpointer_test_file.json"
+
+type testCase struct {
+	description     string
+	after           func(t *testing.T)
+	newCheckpointer func(t *testing.T) (Checkpointer, error)
+}
+
+func CheckFileExist(path string) bool {
+	_, err := os.Stat(path)
+	return !os.IsNotExist(err)
+}
+
+func setupSuite(t *testing.T) func(t *testing.T) {
+	if !CheckFileExist(path) {
+		file, err := os.Create(path)
+		require.NoError(t, err)
+		defer file.Close()
+	}
+	return func(t *testing.T) {
+		if CheckFileExist(path) {
+			err := os.Remove(path)
+			require.NoError(t, err)
+		}
+	}
+}
+
+func assertState(t *testing.T, checkpointer Checkpoint, blocknumber uint64, transactionID string) {
+	require.Equal(t, blocknumber, checkpointer.BlockNumber())
+	require.Equal(t, transactionID, checkpointer.TransactionID())
+}
+
+func TestCheckpointer(t *testing.T) {
+
+	teardownSuite := setupSuite(t)
+	defer teardownSuite(t)
+
+	testCases := []testCase{
+		{
+			description: "In-memory",
+			after:       func(t *testing.T) {},
+			newCheckpointer: func(t *testing.T) (Checkpointer, error) {
+				inmemorCheckpointer := new(InMemoryCheckpointer)
+				return inmemorCheckpointer, nil
+			},
+		},
+		{
+			description: "File",
+			after: func(t *testing.T) {
+				err := os.Remove(path)
+				require.NoError(t, err)
+			},
+			newCheckpointer: func(*testing.T) (Checkpointer, error) {
+				fileCheckpointer, err := NewFileCheckpointer(path)
+				require.NoError(t, err)
+				return fileCheckpointer, err
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+
+		t.Run(tc.description+" :Initializes default checkpointer state when no checkpointer already exist", func(t *testing.T) {
```

Comment:
Rather than adding the testCase description to every test, it might be simpler to wrap them all in an enclosing test run, so all your tests for a specific checkpointer are grouped together
```suggestion
	for _, tc := range testCases {
		t.Run(tc.description, func(t *testing.T) {
			t.Run("Initializes default checkpointer state when no checkpointer already exist", func(t *testing.T) {
```